diff -Naur orig-jenkinsapi/Makefile fixed-jenkinsapi/Makefile
--- orig-jenkinsapi/Makefile	2020-02-16 17:41:39.404723443 -0700
+++ fixed-jenkinsapi/Makefile	2020-02-16 17:42:30.377108829 -0700
@@ -1,7 +1,4 @@
-.PHONY: test lint tox coverage dist
-
-test:
-	py.test-3 -sv jenkinsapi_tests
+.PHONY: lint tox dist
 
 lint:
 	pycodestyle
@@ -13,5 +10,3 @@
 dist:
 	python3 setup.py sdist bdist_wheel
 
-coverage:
-	py.test-3 -sv --cov=jenkinsapi --cov-report=term-missing --cov-report=xml jenkinsapi_tests
