diff -Naur debian-scm/jenkinsapi/credential.py install-fix/jenkinsapi/credential.py
--- debian-scm/jenkinsapi/credential.py	2020-02-16 16:46:46.767307926 -0700
+++ install-fix/jenkinsapi/credential.py	2020-10-27 21:03:49.025512695 -0600
@@ -83,7 +83,7 @@
 
     def __init__(self, cred_dict):
         jenkins_class = 'com.cloudbees.plugins.credentials.impl.UsernamePasswordCredentialsImpl'
-        super(UsernamePasswordCredential, self).__init__(cred_dict, jenkins_class)
+        super().__init__(cred_dict, jenkins_class)
         if 'typeName' in cred_dict:
             username = cred_dict['displayName'].split('/')[0]
         else:
@@ -123,7 +123,7 @@
             'password': self.password,
             'description': self.description
         }
-        return super(UsernamePasswordCredential, self)._get_attributes_xml(data)
+        return super()._get_attributes_xml(data)
 
 
 class SecretTextCredential(Credential):
@@ -146,7 +146,7 @@
 
     def __init__(self, cred_dict):
         jenkins_class = 'org.jenkinsci.plugins.plaincredentials.impl.StringCredentialsImpl'
-        super(SecretTextCredential, self).__init__(cred_dict, jenkins_class)
+        super().__init__(cred_dict, jenkins_class)
         self.secret = cred_dict.get('secret', None)
 
     def get_attributes(self):
@@ -179,7 +179,7 @@
             'secret': self.secret,
             'description': self.description
         }
-        return super(SecretTextCredential, self)._get_attributes_xml(data)
+        return super()._get_attributes_xml(data)
 
 
 class SSHKeyCredential(Credential):
@@ -214,7 +214,7 @@
 
     def __init__(self, cred_dict):
         jenkins_class = 'com.cloudbees.jenkins.plugins.sshcredentials.impl.BasicSSHUserPrivateKey'
-        super(SSHKeyCredential, self).__init__(cred_dict, jenkins_class)
+        super().__init__(cred_dict, jenkins_class)
         if 'typeName' in cred_dict:
             username = cred_dict['displayName'].split(' ')[0]
         else:
@@ -291,7 +291,7 @@
             'privateKeySource': self.attrs,
             'passphrase': self.passphrase,
         }
-        return super(SSHKeyCredential, self)._get_attributes_xml(data)
+        return super()._get_attributes_xml(data)
 
 
 class AmazonWebServicesCredentials(Credential):
@@ -317,7 +317,7 @@
 
     def __init__(self, cred_dict):
         jenkins_class = 'com.cloudbees.jenkins.plugins.awscredentials.AWSCredentialsImpl'
-        super(AmazonWebServicesCredentials, self).__init__(cred_dict, jenkins_class)
+        super().__init__(cred_dict, jenkins_class)
 
         self.access_key = cred_dict['accessKey']
         self.secret_key = cred_dict['secretKey']
@@ -360,4 +360,4 @@
             'iamMfaSerialNumber': self.iam_mfa_serial_number,
             'description': self.description
         }
-        return super(AmazonWebServicesCredentials, self)._get_attributes_xml(data)
+        return super()._get_attributes_xml(data)
diff -Naur debian-scm/jenkinsapi/credentials.py install-fix/jenkinsapi/credentials.py
--- debian-scm/jenkinsapi/credentials.py	2020-02-16 16:46:46.767307926 -0700
+++ install-fix/jenkinsapi/credentials.py	2020-10-27 21:04:33.021889279 -0600
@@ -100,8 +100,8 @@
             except JenkinsAPIException as jae:
                 raise JenkinsAPIException('Latest version of Credentials '
                                           'plugin is required to be able '
-                                          'to create credentials. '
-                                          'Original exception: %s' % str(jae))
+                                          'to create credentials. ') from jae
+
         else:
             cred_id = self[description].credential_id
             credential.credential_id = cred_id
@@ -117,8 +117,7 @@
             except JenkinsAPIException as jae:
                 raise JenkinsAPIException('Latest version of Credentials '
                                           'plugin is required to be able '
-                                          'to update credentials. '
-                                          'Original exception: %s' % str(jae))
+                                          'to update credentials. ') from jae
 
         self.poll()
         self.credentials = self._data['credentials']
@@ -145,8 +144,7 @@
         except JenkinsAPIException as jae:
             raise JenkinsAPIException('Latest version of Credentials '
                                       'required to be able to create '
-                                      'credentials. Original exception: %s'
-                                      % str(jae))
+                                      'credentials.') from jae
         self.poll()
         self.credentials = self._data['credentials']
         if description in self:
diff -Naur debian-scm/jenkinsapi/fingerprint.py install-fix/jenkinsapi/fingerprint.py
--- debian-scm/jenkinsapi/fingerprint.py	2020-02-16 16:46:46.767307926 -0700
+++ install-fix/jenkinsapi/fingerprint.py	2020-10-27 20:59:57.595597808 -0600
@@ -96,14 +96,14 @@
     def validate(self):
         try:
             assert self.valid()
-        except AssertionError:
+        except AssertionError as exc:
             raise ArtifactBroken(
                 "Artifact %s seems to be broken, check %s" %
-                (self.id_, self.baseurl))
-        except requests.exceptions.HTTPError:
+                (self.id_, self.baseurl)) from exc
+        except requests.exceptions.HTTPError as exc:
             raise ArtifactBroken(
                 "Unable to validate artifact id %s using %s" %
-                (self.id_, self.baseurl))
+                (self.id_, self.baseurl)) from exc
         return True
 
     def get_info(self):
diff -Naur debian-scm/jenkinsapi/jenkinsbase.py install-fix/jenkinsapi/jenkinsbase.py
--- debian-scm/jenkinsapi/jenkinsbase.py	2020-02-16 16:46:46.767307926 -0700
+++ install-fix/jenkinsapi/jenkinsbase.py	2020-10-27 20:58:54.431099252 -0600
@@ -84,9 +84,9 @@
             response.raise_for_status()
         try:
             return ast.literal_eval(response.text)
-        except Exception:
+        except Exception as exc:
             logger.exception('Inappropriate content found at %s', url)
-            raise JenkinsAPIException('Cannot parse %s' % response.content)
+            raise JenkinsAPIException('Cannot parse %s' % response.content) from exc
 
     def pprint(self):
         """
diff -Naur debian-scm/jenkinsapi/job.py install-fix/jenkinsapi/job.py
--- debian-scm/jenkinsapi/job.py	2020-02-16 16:46:46.775307980 -0700
+++ install-fix/jenkinsapi/job.py	2020-10-27 20:58:21.502844386 -0600
@@ -89,7 +89,7 @@
         return branches
 
     def poll(self, tree=None):
-        data = super(Job, self).poll(tree=tree)
+        data = super().poll(tree=tree)
         if not tree and not self.jenkins.lazy:
             self._data = self._add_missing_builds(self._data)
 
@@ -405,16 +405,16 @@
             self._revmap = self.get_revision_dict()
         try:
             return self._revmap[revision]
-        except KeyError:
-            raise NotFound("Couldn't find a build with that revision")
+        except KeyError as exc:
+            raise NotFound("Couldn't find a build with that revision") from exc
 
     def get_build(self, buildnumber):
         assert isinstance(buildnumber, int)
         try:
             url = self.get_build_dict()[buildnumber]
             return Build(url, buildnumber, job=self)
-        except KeyError:
-            raise NotFound('Build #%s not found' % buildnumber)
+        except KeyError as exc:
+            raise NotFound('Build #%s not found' % buildnumber) from exc
 
     def delete_build(self, build_number):
         """
@@ -428,8 +428,8 @@
             url = "%s/doDelete" % url
             self.jenkins.requester.post_and_confirm_status(url, data='')
             self.jenkins.poll()
-        except KeyError:
-            raise NotFound('Build #%s not found' % build_number)
+        except KeyError as exc:
+            raise NotFound('Build #%s not found' % build_number) from exc
 
     def get_build_metadata(self, buildnumber):
         """
@@ -443,8 +443,8 @@
         try:
             url = self.get_build_dict()[buildnumber]
             return Build(url, buildnumber, job=self, depth=0)
-        except KeyError:
-            raise NotFound('Build #%s not found' % buildnumber)
+        except KeyError as exc:
+            raise NotFound('Build #%s not found' % buildnumber) from exc
 
     def __delitem__(self, build_number):
         self.delete_build(build_number)
diff -Naur debian-scm/jenkinsapi/node.py install-fix/jenkinsapi/node.py
--- debian-scm/jenkinsapi/node.py	2020-02-16 16:46:46.775307980 -0700
+++ install-fix/jenkinsapi/node.py	2020-10-27 20:54:01.232984206 -0600
@@ -118,10 +118,10 @@
                 credential = self.jenkins.credentials[
                     na['credential_description']
                 ]
-            except KeyError:
+            except KeyError as exc:
                 raise JenkinsAPIException('Credential with description "%s"'
                                           ' not found'
-                                          % na['credential_description'])
+                                          % na['credential_description']) from exc
 
             retries = na['max_num_retries'] if 'max_num_retries' in na else ''
             re_wait = na['retry_wait_time'] if 'retry_wait_time' in na else ''
diff -Naur debian-scm/jenkinsapi/nodes.py install-fix/jenkinsapi/nodes.py
--- debian-scm/jenkinsapi/nodes.py	2020-02-16 16:46:46.775307980 -0700
+++ install-fix/jenkinsapi/nodes.py	2020-10-27 20:50:45.423835163 -0600
@@ -75,8 +75,8 @@
             nodename = item['displayName']
             try:
                 yield nodename, self._make_node(nodename)
-            except Exception:
-                raise JenkinsAPIException('Unable to iterate nodes')
+            except Exception as exc:
+                raise JenkinsAPIException('Unable to iterate nodes') from exc
 
     def items(self):
         """
@@ -94,8 +94,8 @@
         for item in self._data['computer']:
             try:
                 yield self._make_node(item['displayName'])
-            except Exception:
-                raise JenkinsAPIException('Unable to iterate nodes')
+            except Exception as exc:
+                raise JenkinsAPIException('Unable to iterate nodes') from exc
 
     def values(self):
         """
diff -Naur debian-scm/jenkinsapi/plugins.py install-fix/jenkinsapi/plugins.py
--- debian-scm/jenkinsapi/plugins.py	2020-02-16 16:46:46.775307980 -0700
+++ install-fix/jenkinsapi/plugins.py	2020-10-27 20:49:44.063536416 -0600
@@ -82,8 +82,8 @@
     def __getitem__(self, plugin_name):
         try:
             return self.get_plugins_dict()[plugin_name]
-        except KeyError:
-            raise UnknownPlugin(plugin_name)
+        except KeyError as exc:
+            raise UnknownPlugin(plugin_name) from exc
 
     def __setitem__(self, shortName, plugin):
         """
diff -Naur debian-scm/jenkinsapi/queue.py install-fix/jenkinsapi/queue.py
--- debian-scm/jenkinsapi/queue.py	2020-02-16 16:46:46.775307980 -0700
+++ install-fix/jenkinsapi/queue.py	2020-10-27 20:48:12.011156888 -0600
@@ -170,11 +170,11 @@
     def get_build_number(self):
         try:
             return self._data['executable']['number']
-        except (KeyError, TypeError):
-            raise NotBuiltYet()
+        except (KeyError, TypeError) as exc:
+            raise NotBuiltYet() from exc
 
     def get_job_name(self):
         try:
             return self._data['task']['name']
-        except KeyError:
-            raise NotBuiltYet()
+        except KeyError as exc:
+            raise NotBuiltYet() from exc
diff -Naur debian-scm/Makefile install-fix/Makefile
--- debian-scm/Makefile	2020-02-16 16:46:46.767307926 -0700
+++ install-fix/Makefile	2020-10-27 20:25:31.785018612 -0600
@@ -1,17 +1,13 @@
-.PHONY: test lint tox coverage dist
-
-test:
-	py.test -sv jenkinsapi_tests
+.PHONY: lint tox dist install
 
 lint:
-	pycodestyle
 	pylint jenkinsapi/*.py
 
 tox:
 	tox
 
 dist:
-	python setup.py sdist bdist_wheel
+	python3 setup.py sdist bdist_wheel
 
-coverage:
-	py.test -sv --cov=jenkinsapi --cov-report=term-missing --cov-report=xml jenkinsapi_tests
+install:
+	python3 setup.py install --prefix=${DESTDIR}/usr
diff -Naur debian-scm/.pc/applied-patches install-fix/.pc/applied-patches
--- debian-scm/.pc/applied-patches	1969-12-31 17:00:00.000000000 -0700
+++ install-fix/.pc/applied-patches	2020-10-27 20:36:14.447239048 -0600
@@ -0,0 +1,5 @@
+python3.patch
+notest.patch
+codestyle.patch
+install.patch
+pylint.patch
diff -Naur debian-scm/.pc/codestyle.patch/Makefile install-fix/.pc/codestyle.patch/Makefile
--- debian-scm/.pc/codestyle.patch/Makefile	1969-12-31 17:00:00.000000000 -0700
+++ install-fix/.pc/codestyle.patch/Makefile	2020-10-27 20:23:40.220077140 -0600
@@ -0,0 +1,12 @@
+.PHONY: lint tox dist
+
+lint:
+	pycodestyle
+	pylint jenkinsapi/*.py
+
+tox:
+	tox
+
+dist:
+	python3 setup.py sdist bdist_wheel
+
diff -Naur debian-scm/.pc/install.patch/Makefile install-fix/.pc/install.patch/Makefile
--- debian-scm/.pc/install.patch/Makefile	1969-12-31 17:00:00.000000000 -0700
+++ install-fix/.pc/install.patch/Makefile	2020-10-27 20:23:40.244077336 -0600
@@ -0,0 +1,11 @@
+.PHONY: lint tox dist
+
+lint:
+	pylint jenkinsapi/*.py
+
+tox:
+	tox
+
+dist:
+	python3 setup.py sdist bdist_wheel
+
diff -Naur debian-scm/.pc/notest.patch/Makefile install-fix/.pc/notest.patch/Makefile
--- debian-scm/.pc/notest.patch/Makefile	1969-12-31 17:00:00.000000000 -0700
+++ install-fix/.pc/notest.patch/Makefile	2020-10-27 20:23:40.200076977 -0600
@@ -0,0 +1,17 @@
+.PHONY: test lint tox coverage dist
+
+test:
+	py.test-3 -sv jenkinsapi_tests
+
+lint:
+	pycodestyle
+	pylint jenkinsapi/*.py
+
+tox:
+	tox
+
+dist:
+	python3 setup.py sdist bdist_wheel
+
+coverage:
+	py.test-3 -sv --cov=jenkinsapi --cov-report=term-missing --cov-report=xml jenkinsapi_tests
diff -Naur debian-scm/.pc/pylint.patch/Makefile install-fix/.pc/pylint.patch/Makefile
--- debian-scm/.pc/pylint.patch/Makefile	1969-12-31 17:00:00.000000000 -0700
+++ install-fix/.pc/pylint.patch/Makefile	2020-10-27 20:24:20.916413374 -0600
@@ -0,0 +1,13 @@
+.PHONY: lint tox dist install
+
+lint:
+	pylint3 jenkinsapi/*.py
+
+tox:
+	tox
+
+dist:
+	python3 setup.py sdist bdist_wheel
+
+install:
+	python3 setup.py install --prefix=${DESTDIR}/usr
diff -Naur debian-scm/.pc/python3.patch/Makefile install-fix/.pc/python3.patch/Makefile
--- debian-scm/.pc/python3.patch/Makefile	1969-12-31 17:00:00.000000000 -0700
+++ install-fix/.pc/python3.patch/Makefile	2020-10-27 20:23:40.172076749 -0600
@@ -0,0 +1,17 @@
+.PHONY: test lint tox coverage dist
+
+test:
+	py.test -sv jenkinsapi_tests
+
+lint:
+	pycodestyle
+	pylint jenkinsapi/*.py
+
+tox:
+	tox
+
+dist:
+	python setup.py sdist bdist_wheel
+
+coverage:
+	py.test -sv --cov=jenkinsapi --cov-report=term-missing --cov-report=xml jenkinsapi_tests
diff -Naur debian-scm/.pc/.quilt_patches install-fix/.pc/.quilt_patches
--- debian-scm/.pc/.quilt_patches	1969-12-31 17:00:00.000000000 -0700
+++ install-fix/.pc/.quilt_patches	2020-10-27 20:23:40.156076618 -0600
@@ -0,0 +1 @@
+/home/ahs3/debian/python-jenkinsapi/debian-scm/debian/patches
diff -Naur debian-scm/.pc/.quilt_series install-fix/.pc/.quilt_series
--- debian-scm/.pc/.quilt_series	1969-12-31 17:00:00.000000000 -0700
+++ install-fix/.pc/.quilt_series	2020-10-27 20:23:40.156076618 -0600
@@ -0,0 +1 @@
+series
diff -Naur debian-scm/.pc/.version install-fix/.pc/.version
--- debian-scm/.pc/.version	1969-12-31 17:00:00.000000000 -0700
+++ install-fix/.pc/.version	2020-10-27 20:23:40.156076618 -0600
@@ -0,0 +1 @@
+2
