--- a/Makefile
+++ b/Makefile
@@ -1,7 +1,7 @@
 .PHONY: test lint tox coverage dist
 
 test:
-	py.test -sv jenkinsapi_tests
+	py.test-3 -sv jenkinsapi_tests
 
 lint:
 	pycodestyle
@@ -11,7 +11,7 @@
 	tox
 
 dist:
-	python setup.py sdist bdist_wheel
+	python3 setup.py sdist bdist_wheel
 
 coverage:
-	py.test -sv --cov=jenkinsapi --cov-report=term-missing --cov-report=xml jenkinsapi_tests
+	py.test-3 -sv --cov=jenkinsapi --cov-report=term-missing --cov-report=xml jenkinsapi_tests
